### DISCUSSION TITLE
How to configure custom maven repository

### DISCUSSION THREAD

**Discussion by bitsInnovate00** [apache/camel-karavan#1226] [Q&A] [answered]
### Ask a question

How can I configure a private maven repository in karavan integration build which will load the necessary custom maven dependencies of the project (private dependencies which are not in maven central repo)?

Currently it will try to get dependencies from maven central repo which will fail.  Are there any properties in application.properties similar to camel.jbang.dependencies where i can configure the private repo

### Variant

None

### Container Management (if applicable)

Kubernetes

### Operating System (if applicable)

Linux

### Version

4.4.0

**Comment by mgubaidullin** [comment]
You need to configure `camel.jbang.repos` in `application.properties`

More info https://camel.apache.org/manual/camel-jbang.html#_using_3rd_party_maven_repositories

**Comment by mgubaidullin** [comment]
And you don't need to add camel components you use to deps in application.properties. Camel-jbang analyses Yaml routes and adds dependencies automagically

**Reply by bitsInnovate00** [reply]
Thank you for the quick reply !!!

I doubt I can use camel.jbang.repos since the repository is authenticated and credentials needs to be passed. So I'm looking for an option to pass the maven settings.xml file. I could find a property camel.jbang.mavenSettings but the doubt is how can I pass this file path since builder is a container image. Is there any option available or do I need to extend the devmode image to include this functionality ?

**Reply by mgubaidullin** [reply, marked as answer]
I think you need make a custom devmode image, and set `karavan.devmode.image` property for Karavan Application
